src/server/cluster/cluster_family.cc

```diff
@@ -764,29 +776,27 @@ std::shared_ptr<OutgoingMigration> ClusterFamily::CreateOutgoingMigration(std::s
 
 void ClusterFamily::DflyMigrateFlow(CmdArgList args, ConnectionContext* cntx) {
   CmdArgParser parser{args};
-  auto [sync_id, shard_id] = parser.Next<uint32_t, uint32_t>();
+  auto [port, shard_id] = parser.Next<uint32_t, uint32_t>();
 
   if (auto err = parser.Error(); err) {
     return cntx->SendError(err->MakeReply());
   }
 
-  VLOG(1) << "Create flow sync_id: " << sync_id << " shard_id: " << shard_id;
+  auto host_ip = cntx->conn()->RemoteEndpointAddress();
 
-  cntx->conn()->SetName(absl::StrCat("migration_flow_", sync_id));
+  VLOG(1) << "Create flow " << host_ip << ":" << port << " shard_id: " << shard_id;
 
-  auto migration = GetOutgoingMigration(sync_id);
+  cntx->conn()->SetName(absl::StrCat("migration_flow_", host_ip, ":", port));
+
+  auto migration = GetIncomingMigration(cntx->conn()->RemoteEndpointAddress(), port);
   if (!migration)
     return cntx->SendError(kIdNotFound);
 
-  cntx->conn()->Migrate(shard_set->pool()->at(shard_id));
-  server_family_->journal()->StartInThread();
+  cntx->sync_dispatch = false;
```

Comment:
please DCHECK it's actually true because with async_dispatch everything will break

Reply:
and add a comment that we do this to be ignored by the dispatch tracker and maybe a todo that we should add a cleaner approach (some kind of separate technical connection flag - like you suggested)